app/scripts/controllers/new_proposal.js

```diff
@@ -8,52 +8,34 @@ angular.module('admissionSystemApp')
       $scope.entireSpecoffer.specoffer = {};
       $scope.entireSpecoffer.specoffer.timePeriodId = valueSendingService.timeperiod;
 
-      SpecofferDictionaryService.getAllDepartments().then(function (departments) {
-        $scope.departmentId = departments;
-      });
-      SpecofferDictionaryService.getSpecoffersTypes().then(function (SpecoffersTypes) {
-        $scope.specofferTypesOptions = SpecoffersTypes;
-      });
-      SpecofferDictionaryService.getEduformTypes().then(function (eduformtypes) {
-         $scope.eduFormTypesOptions = eduformtypes;
-      });
-
-      SpecofferDictionaryService.getTimePeriodCourseIds().then(function (timePeriodCourseIds) {
-        $scope.timePeriodCourseId = timePeriodCourseIds;
-      });
-
-  $scope.brosweOrEditSpecoffer = function  (specofferId) {
-    SpecoffersService.getEntireSpecoffer(specofferId).then(function(res){
-      _.merge($scope.entireSpecoffer.subjects, res.subjects);
-      _.merge($scope.entireSpecoffer.benefits, res.benefits);
-      _.merge($scope.entireSpecoffer.specoffer, res.specoffer);
+    $q.all([
+      SpecofferDictionaryService.getAllDepartments(),
+      SpecofferDictionaryService.getSpecoffersTypes(),
+      SpecofferDictionaryService.getEduformTypes(),
+      SpecofferDictionaryService.getTimePeriodCourseIds()
+    ])
+    .then(function(promisesResult) {
+      $scope.departmentId = promisesResult[0];
```

Comment:
As I can see, promisesResult[0] isn't any id, maybe better name for variable something like departmentList